app/controllers/api/v1/items_controller.rb

```diff
@@ -29,4 +29,10 @@ def destroy
   def item_params
     params.require(:item).permit(:name, :description, :unit_price, :merchant_id)
   end
+
+  def items_selection
+    return Item.all if params["merchant_id"] == nil
+    return Item.where(merchant_id: params["merchant_id"]) if params["merchant_id"]
+  end
```

Comment:
@dionew1 @BrianZanti, I'm not sure if this is appropriate logic for the rails-engine. Do I need to create a model method to facilitate this process?

Reply:
I think this is okay logic to be in the controller. You may need to pass items_selection an argument of params so that the method has access to it.